### ISSUE TITLE
Event for Level 1

### ISSUE THREAD

**Issue by pwintthiri** [nus-cs2030/2122-s2#355]
Hi, for Event in level 1, 
`Event event = new EventStub(new Customer(1, 0.5), 3.7)`
seems to suggest that Event can be instantiated and hence it's a class. 
But in level 2, Event is supposed to be an interface. 

So I'm confused about how to go about creating EventStub for level 1. Is it possible for there to be one Event class and one Event interface? Or I might be approaching it the wrong way so any other advice would be helpful too, thank you!

![Screenshot 2022-04-17 at 12 54 40 PM](https://user-images.githubusercontent.com/95198407/163701016-23889b94-c795-4390-8012-bc0696042b1b.png)

**Comment by haoyangw** [comment]
> seems to suggest that Event can be instantiated and hence it's a class.

Actually no, this is an example of polymorphism. You can always store an instance of a class implementing an interface within a variable of the interface's type, like in this case. Event can be either a class or interface, it doesn't matter. As long as EventStub either implements Event(interface) or extends Event(class).

I don't think it's possible to have one Event class and one Event interface. You may choose to write an Event interface or an Event (abstract) class, both are fine.

**Comment by kkinmunn** [comment]
To add on with what @haoyangw has said, it is not possible to have an interface and a class with the same name (in the same directory/folder) as you will have a "duplicate class" error in java. You can read a bit more here: 
https://stackoverflow.com/questions/43164923/java-class-and-interface-name-collision#:~:text=You%20can't%20have%20a,language%20doesn't%20allow%20it.
Hope this helps!

**Comment by yjjjiajie** [comment]
for me, i made Event an abstract class and it worked for the whole project. But I do have friends making Event an interface and passed the project too. So ultimately it really depends on you. The LHS can be an interface. However, if the RHS shows "new Event()", that is when you know it cannot be an interface.

**Comment by terrrence** [comment]
hi! my event is an interface so my subsequent event classes (eventstub, arrive etc) implements event.

**Comment by alwinngjw** [comment]
Hi, i personally used an interface for my Event class and my EventStub implements it.

**Comment by lukasleeyo** [comment]
Hi OP, you can make your ``Event`` an Interface or an abstract class. Your ``EventStub`` only needs to implement the Event if it's an interface, and extends the Event if it's an abstract class.


TLDR:
**If you make Event an interface**
```java
interface Event {
your code...
}

public class EventStub implements Event {
your code...
}
```
**If you make Event an abstract class**
```java
public abstract class Event {
your code...
}

public class EventStub extends Event {
your code...
}
```

**Comment by summerthia02** [comment]
I used an interface so all my other events implements Event

**Comment by hhujiale** [comment]
I made it as an abstract class and other events extends Event, I think its also fine
